plugins/modules/provision_workflow_manager.py

```diff
@@ -933,56 +943,56 @@ class instance for further use.
                     self.log(self.msg, "ERROR")
                     self.status = "failed"
                     return self
+
+            if to_provisioning is True:
+                try:
+                    response = self.dnac_apply['exec'](
+                        family="sda",
+                        function="provision_wired_device",
+                        op_modifies=True,
+                        params=self.want["prov_params"],
+                    )
+                    self.log("Provisioning response collected from 'provision_wired_device' API is: {0}".format(response), "DEBUG")
+                except Exception as e:
+                    self.msg = "Error in provisioning due to {0}".format(str(e))
+                    self.log(self.msg, "ERROR")
+                    self.status = "failed"
+                    return self
+
             else:
-                if self.validated_config[0].get("provisioning") is True:
-                    try:
-                        response = self.dnac_apply['exec'](
-                            family="sda",
-                            function="provision_wired_device",
-                            op_modifies=True,
-                            params=self.want["prov_params"],
-                        )
-                        self.log("Provisioning response collected from 'provision_wired_device' API is: {0}".format(response), "DEBUG")
-                    except Exception as e:
-                        self.msg = "Error in provisioning due to {0}".format(str(e))
-                        self.log(self.msg, "ERROR")
-                        self.status = "failed"
-                        return self
+                uuid = self.get_device_id()
+                if self.is_device_assigned_to_site(uuid) is True:
+                    self.result["changed"] = False
+                    self.result['msg'] = "Device is already assigned to the desired site"
```

Comment:
Add the device ip so that log message will be clear.

Reply:
updated the code